integration-tests/src/test/java/io/quarkus/qe/extensions/QuarkusAppsCatalogDeploymentExtension.java

```diff
@@ -0,0 +1,80 @@
+package io.quarkus.qe.extensions;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+
+import io.quarkus.qe.containers.EnricherServiceContainer;
+import io.quarkus.qe.containers.PostgreSqlContainer;
+import io.quarkus.qe.containers.RestApiServiceContainer;
+import io.quarkus.qe.containers.StorageServiceContainer;
+
+public class QuarkusAppsCatalogDeploymentExtension implements BeforeAllCallback, AfterAllCallback {
+
+    public static final String DATABASE_ALIAS = "database";
+    public static final String KAFKA_ALIAS = "kafka";
+
+    private Network network;
+    private PostgreSqlContainer database;
+    private KafkaContainer kafka;
+    private StorageServiceContainer storageService;
+    private RestApiServiceContainer restApiService;
+    private EnricherServiceContainer enricherService;
+
+    @SuppressWarnings({ "resource", "deprecation" })
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+        network = Network.newNetwork();
+        database = new PostgreSqlContainer().withNetwork(network).withNetworkAliases(DATABASE_ALIAS);
+        database.start();
+
+        kafka = new KafkaContainer().withNetwork(network).withNetworkAliases(KAFKA_ALIAS);
+        kafka.start();
+
+        storageService = new StorageServiceContainer();
+        storageService.withNetwork(network);
+        storageService.start();
+
+        enricherService = new EnricherServiceContainer();
+        enricherService.withNetwork(network);
+        enricherService.start();
+
+        restApiService = new RestApiServiceContainer();
+        restApiService.withNetwork(network);
+        restApiService.start();
```

Comment:
All of this `components.start()` could be replaces by 

`Startables.deepStart(Stream.of(components1, component2....)).join();`

Reply:
Changed.